src/versions.rs

```diff
@@ -224,6 +236,46 @@ fn upgrade_v0_pin(name: &str, raw_pin: &mut Map<String, Value>) -> Result<()> {
     Ok(())
 }
 
+/* v5→v6. This upgrade changes the hashes of git and git-release pins to use SRI hashes instead of
+ * raw sha256 hashes.
+ */
+fn upgrade_v5_pin(name: &str, raw_pin: &mut Map<String, Value>) -> Result<()> {
+    let pin: Pin = serde_json::from_value(serde_json::Value::Object(raw_pin.clone()))?;
```

Comment:
This is an acceptable solution for now, however it is not entirely future proof as the `Pin` type used here will implicitly bitrot with following migrations. This can be worked around with the next migration, however, if you could find a stringly-typed solution that simply replaces "hash" fields that would be favorable.

(The general migration solution is still lacking, however the only way I could see building a fully typed solution would be by duplicating the entire `Pin` type with all its variants for every version migration and ugh)